### PR TITLE
feat: application support

### LINKED ISSUE #31: Allow multiple dapps for a given author
Currently, each author can only store one dapp under their namespace. They should be able to specify a name in order to support many dapps per account

### PR DESCRIPTION
**Pull request by andy-haynes**
This PR adds support for applications, a logical grouping of files that allows a single author to upload and manage assets for multiple dapps.

Fixes #31 

First time app deploy:
- reinstall & rebuild
- in `packages/react/package.json`, change the `nearDeployConfig` values for your application (leave `filestoreContract` unless you're deploying your own)
- in `packages/react`, run `pnpm run deploy` to deploy the application for the first time; this will output progress and debugging values
- to delete the application and unregister user storage, run `pnpm delete-app`
- for a clean redeploy, run `pnpm clean-deploy` (note that this only works if you already have an app deployed, since `delete-app` currently breaks if the app does not already exist)

### CODE REVIEW

File: packages/cdn/src/index.ts
Comment (pavelisnear) [inline packages/cdn/src/index.ts:10]: Looks like the list of arguments is growing, I think it might make sense to avoid [connascence of position](https://connascence.io/position.html) by creating an object param as a config of the query. How does the similar in `provider.query` call look to you?

File: packages/filestore-contract/src/partitioned.ts
Comment (pavelisnear) [inline packages/filestore-contract/src/partitioned.ts:58]: I guess we should do a check on the `app` argument before calling this function, so that `app` would always be defined as the interface suggests. Otherwise, the next next lines would also require an optional chaining.

File: packages/react/package.json
Comment (pavelisnear) [inline packages/react/package.json:37]: Shall we call this `deployConfig` to avoid `bos` acronym?

**Review by pavelisnear — APPROVED** [review]
Great job! 
I really like the new `Application` approach with the versioning!

File: packages/cdn/src/index.ts
Comment (andy-haynes) [reply packages/cdn/src/index.ts:10]: Yup this whole thing needs a rework, it's really hacky.

File: packages/react/package.json
Comment (andy-haynes) [reply packages/react/package.json:37]: I'd prefer that it's clearly a Near-related field to make it clear it's user-defined and avoid any potential collisions.

**Comment by mpeterdev** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@andy-haynes Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/andy-haynes?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/andy-haynes?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/andy-haynes)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/andy-haynes)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by mpeterdev** [comment]
@race-of-sloths score 8

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@mpeterdev, thank you for scoring this pull request in the Race of Sloths!

File: packages/react/package.json
Comment (mpeterdev) [reply packages/react/package.json:37]: yeah let's please avoid any `bos` references 😅 

`nearDeployConfig`?

**Comment by mpeterdev** [comment]
build failing due to missing `@near-js/util` in `packages/presets`

File: packages/paid-storage-contract/src/contract.ts
Comment (mpeterdev) [inline packages/paid-storage-contract/src/contract.ts:439]: my understanding is that it is best practice to use predecessor instead of signer unless you have a specific flow that requires signer e.g. a marketplace

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 8!
You have received 100 (80 base + 10 weekly bonus + 10 monthly bonus) Sloth points for this contribution


Congratulations @andy-haynes! Your PR was highly scored and you completed another monthly streak! To keep your monthly streak make another pull request next month and get 8+ score for it

**Pull request merged**
